entity-service-impl/src/main/java/org/hypertrace/entity/query/service/EntityAttributeMapping.java

```diff
@@ -52,6 +54,19 @@
         .or(() -> this.calculateDocStorePathFromAttributeId(requestContext, attributeId));
   }
 
+  public boolean isMultiValued(RequestContext requestContext, String attributeId) {
+    return requestContext.call(
+        () ->
+            this.attributeClient
+                .get(attributeId)
+                .filter(metadata -> metadata.getSourcesList().contains(AttributeSource.EDS))
+                .map(AttributeMetadata::getValueKind)
+                .map(valueKind -> (AttributeKind.TYPE_STRING_ARRAY == valueKind))
+                .onErrorComplete()
+                .defaultIfEmpty(false)
+                .blockingGet());
```

Comment:
I am not exactly sure, what's the correct norm, but shouldn't consumers of `isMultiValued` function call `blockingGet()`?

Reply:
Generally, I prefer to leave things async by default and have the consumer decide (which is why attribute client is that way) but when I added the AS integration in this class, since it's a local impl class I chose to block so as to not introduce Rx across the service more broadly.